app/javascript/packages/document-capture/components/document-capture-abandon.tsx

```diff
@@ -0,0 +1,133 @@
+        {children}
+      </Link>
+    ),
+    strong: ({ children }) => <strong>{children}</strong>,
+  });
+}
+
+export interface DocumentCaptureAbandonProps {
+  navigate?: Navigate;
+}
+
+function DocumentCaptureAbandon({ navigate }: DocumentCaptureAbandonProps) {
+  const { t } = useI18n();
+  const { trackEvent } = useContext(AnalyticsContext);
+  const { currentStep, exitURL, cancelURL } = useContext(FlowContext);
+  const { name: spName } = useContext(ServiceProviderContext);
+  const appName = getConfigValue('appName');
+  const header = <h2 className="h3">{t('doc_auth.exit_survey.header')}</h2>;
+
+  const content = (
+    <p>
+      {formatContentHtml({
+        msg: spName?.trim()
+          ? t('doc_auth.exit_survey.content_html', {
+              sp_name: spName,
+              app_name: appName,
+            })
+          : t('doc_auth.exit_survey.content_nosp_html', {
+              app_name: appName,
+            }),
+        url: addSearchParams(spName?.trim() ? exitURL : cancelURL, {
+          step: currentStep,
+          location: 'optional_question',
+        }),
+      })}
+    </p>
+  );
+  const optionalTag = (
+    <Tag isBig={false} isInformative>
+      {t('doc_auth.exit_survey.optional.tag', { app_name: appName })}
+    </Tag>
+  );
+  const optionalText = (
+    <p className="margin-top-3">
+      <strong>{t('doc_auth.exit_survey.optional.content', { app_name: appName })}</strong>
+    </p>
+  );
+
+  const idTypeLabels = [
+    t('doc_auth.exit_survey.optional.id_types.us_passport'),
+    t('doc_auth.exit_survey.optional.id_types.resident_card'),
+    t('doc_auth.exit_survey.optional.id_types.military_id'),
+    t('doc_auth.exit_survey.optional.id_types.tribal_id'),
+    t('doc_auth.exit_survey.optional.id_types.voter_registration_card'),
+    t('doc_auth.exit_survey.optional.id_types.other'),
+  ];
+
+  const allIdTypeOptions = [
+    { name: 'us_passport', checked: false },
+    { name: 'resident_card', checked: false },
+    { name: 'military_id', checked: false },
+    { name: 'tribal_id', checked: false },
+    { name: 'voter_registration_card', checked: false },
+    { name: 'other', checked: false },
+  ];
+
+  const [idTypeOptions, setIdTypeOptions] = useState(allIdTypeOptions);
+
+  const updateCheckStatus = (index: number) => {
+    setIdTypeOptions(
+      idTypeOptions.map((idOption, currentIndex) =>
+        currentIndex === index ? { ...idOption, checked: !idOption.checked } : { ...idOption },
+      ),
+    );
+  };
+
+  const checkboxes = (
+    <>
+      {idTypeOptions.map((idType, idx) => (
+        <Checkbox
+          key={idType.name}
+          name={idType.name}
+          value={idType.name}
+          label={idTypeLabels[idx]}
+          onChange={() => updateCheckStatus(idx)}
+        />
+      ))}
+    </>
+  );
+
+  const handleExit = () => {
+    trackEvent('IdV: exit optional questions', { ids: idTypeOptions });
+    const url = spName?.trim() ? exitURL : cancelURL;
```

Comment:
What is the `trim()` meant to account for here?

Reply:
@aduth , this is just in case we have empty spName.

Reply:
It's the `trim` that I'm most curious about to me. It would imply to me we're expecting an `spName` that is not empty, but only contains whitespace (e.g. `'   '`). I don't think we're expecting that?

In other words, would it not be enough to do this?

```suggestion
    const url = spName ? exitURL : cancelURL;
```

Reply:
@aduth , was thinking about a little protection. Anyway, removed it.